plugins/c9.ide.ace.keymaps/cli.js

```diff
@@ -187,6 +187,122 @@ define(function(require, exports, module) {
             }
         };
         
+        function getCmdLineRange(editor, range, parts) {
+            if (!range) {
+                var cursor = editor.selection.getCursor();
+                return [cursor.row, cursor.row];
+            }
+            
+            range = range.split(',');
+            if (range.length === 1) {
+                return [0, parts.length - 1];
+            }
+
+            if (range[0] === "'<") {
+                var selRange = editor.getSelectionRange();
+                return [selRange.start.row, selRange.end.row];
+            } else {
+                var result = [];
+                for (let i = 0; i < range.length; i++) {
+                    var v = range[i];
+                    var rangeVal = v === '.' ?
+                        editor.selection.getCursor().row
+                        : v === '$' ?
+                            parts.length - 1
+                            : v[0] === '+' ?
+                                result[0] + parseInt(v.slice(1), 10)
+                                : parseInt(v, 10);
+                    if (rangeVal >= parts.length) {
+                        rangeVal = parts.length - 1;
+                    }
+                    
+                    result.push(rangeVal);
+                }
+                
+                return result;
+            }
+        }
+        
+        cliCmds[':'].reCommands = {
+            /**
+             * @see {@link http://vim.wikia.com/wiki/Search_and_replace|Vim wiki - sed}
+             */
+            'sed': {
+                regex: /^(%|'<,'>|(?:\d+|\.),(?:\+?\d+|\$|\.))?s(\/|#)(.*?)\2(.*?)\2([giIc]*)$/,
+                action: function (editor, cmd, data) {
+                    var pattern = (
```

Comment:
have you seen the implementation in vim mode, if yes how does this compare with it?  https://github.com/ajaxorg/ace/blob/master/lib/ace/keyboard/vim.js#L5266